packages/headless/src/controllers/url-manager/headless-url-manager.ts

```diff
@@ -0,0 +1,117 @@
+import {Schema, StringValue} from '@coveo/bueno';
+import {Engine} from '../../app/headless-engine';
+import {restoreSearchParameters} from '../../features/search-parameters/search-parameter-actions';
+import {SearchParametersState} from '../../state/search-app-state';
+import {validateInitialState} from '../../utils/validate-payload';
+import {buildController, Controller} from '../controller/headless-controller';
+import {executeSearch} from '../../features/search/search-actions';
+import {ConfigurationSection} from '../../state/state-sections';
+import {buildSearchParameterSerializer} from '../../features/search-parameters/search-parameter-serializer';
+import {buildSearchParameterManager} from '../search-parameter-manager/headless-search-parameter-manager';
+import {logParametersChange} from '../../features/search-parameters/search-parameter-analytics-actions';
+import {initialSearchParameterSelector} from '../../features/search-parameters/search-parameter-selectors';
+
+export interface UrlManagerProps {
+  /**
+   * The initial state that should be applied to the `UrlManager` controller.
+   */
+  initialState: UrlManagerInitialState;
+}
+
+export interface UrlManagerInitialState {
+  /**
+   * The part of the url containing the parameters affecting the search response.
+   */
+  fragment: string;
+}
+
+const initialStateSchema = new Schema<Required<UrlManagerInitialState>>({
+  fragment: new StringValue(),
+});
+
+/**
+ * The `UrlManager` controller allows managing an url which affect the search response.
```

Comment:
From reading the doc comments in this file, it's my impression that this controller takes a portion of a given url and uses it to run a new query. But it's not automatic, right? A user of headless would have to write their own code to watch the url for changes, and call `UrlManager.synchronize` each time, correct?
If this is the case, I'd go with something closer to "The `UrlManager` controller can parse urls to extract search parameters which affect the search response."

Reply:
Yes perfect! Thanks 😄